src/main/java/core/basesyntax/service/impl/BalanceAction.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ActionHandler;
+
+public class BalanceAction implements ActionHandler {
+    public void apply(String fruit, int quantity) {
+        Storage.setBalance(fruit, quantity);
+    }
```

Comment:
It looks like the `apply` method in `BalanceAction` class directly manipulates the `Storage`. Following SOLID principles, especially the Single Responsibility Principle, this class should not be responsible for modifying the storage directly. Consider having a service that interacts with the Storage and let this class call that service instead.